object_store_factory/src/local.rs

```diff
@@ -46,6 +53,16 @@ impl LocalConfig {
     pub fn build_local_storage(
         &self,
     ) -> Result<Arc<dyn ObjectStore>, object_store::Error> {
+        if self.create_dir {
+            // `LocalFileSystem::new_with_prefix` will error out unless the path fully exists
+            fs::create_dir_all(self.data_dir.clone()).map_err(|e| {
```

Comment:
Does this error if the directory already exists? i.e. if we have two writes, will the second one break?